nms/packages/magmalte/app/views/subscriber/SubscriberAddDialog.js

```diff
@@ -360,35 +281,50 @@ type EditProps = {
   editTable: $Keys<typeof EditTableType>,
 };
 
+type ActionDialogProps = {
+  open: boolean,
+  onClose: () => void,
+  editProps?: EditProps,
+  onSave?: (
+    subscribers: Array<SubscriberInfo>,
+    selectedSubscribers?: Array<string>,
+  ) => void,
+  error?: string,
+  subscriberAction: SubscriberActionType,
+};
+
 type DialogProps = {
   open: boolean,
   onClose: () => void,
   editProps?: EditProps,
-  onSave?: (subscribers: Array<SubscriberInfo>) => void,
+  onSave?: (
+    subscribers: Array<SubscriberInfo>,
+    selectedSubscribers?: Array<string>,
+  ) => void,
```

Comment:
It's very unclear to me why there are two different types here, just differing by the presence of `subscriberAction` field. I also see that these `Props` types are reused. I'd suggest:
- each `Props` type is only used for one component. Reusable types can be hard to interpret
- avoid optional fields - I suspect `onSave` shouldn't be optional, but it's fine for `error`
- add comments to explain what the `Props` type are for, or have JSDoc comments on the component that uses the `Props` type.

Reply:
You have a lot of different `onSave` fields throughout this file. Change the function signature so there aren't optional fields. If `onSave` is use for add AND edit AND delete, add type definitions, one type definition for `onAdd`, one for `onEdit`, and one for `onDelete`